File: Prova_Trim2_Shaikoski/JS/estacoes.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const meses = {
  "Janeiro": 1,
  "Fevereiro": 2,
  "Março": 3,
  "Abril": 4,
  "Maio": 5,
  "Junho": 6,
  "Julho": 7,
  "Agosto": 8,
  "Setembro": 9,
  "Outubro": 10,
  "Novembro": 11,
  "Dezembro": 12
};

function getEstacao(mes, dia) {
  if ((mes === 12 && dia >= 21) || (mes === 1 && dia < 21) || (mes === 2)) {
    return "Verão";
  } else if ((mes === 3 && dia >= 21) || (mes === 4) || (mes === 5) || (mes === 6 && dia < 21)) {
    return "Outono";
  } else if ((mes === 6 && dia >= 21) || (mes === 7) || (mes === 8) || (mes === 9 && dia < 23)) {
    return "Inverno";
  } else if ((mes === 9 && dia >= 23) || (mes === 10) || (mes === 11) || (mes === 12 && dia < 21)) {
    return "Primavera";
  }
}

rl.question('Digite o número do mês: ', (mesNumero) => {
  rl.question('Digite o dia do mês: ', (diaNumero) => {
    mesNumero = parseInt(mesNumero);
    diaNumero = parseInt(diaNumero);

    for (const mes in meses) {
      if (meses[mes] === mesNumero) {
        const estacao = getEstacao(mesNumero, diaNumero);
        console.log(`O mês é ${mes} e a estação é ${estacao}`);
        break;
      }
    }

    rl.close();
  });
});

File: Prova_Trim2_Shaikoski/Python/estacoes.py
meses = {
    "Janeiro": 1,
    "Fevereiro": 2,
    "Março": 3,
    "Abril": 4,
    "Maio": 5,
    "Junho": 6,
    "Julho": 7,
    "Agosto": 8,
    "Setembro": 9,
    "Outubro": 10,
    "Novembro": 11,
    "Dezembro": 12
}

def get_estacao(mes, dia):
    if (mes == 12 and dia >= 21) or (mes == 1 and dia < 21) or (mes == 2):
        return "Verão"
    elif (mes == 3 and dia >= 21) or (mes == 4) or (mes == 5) or (mes == 6 and dia < 21):
        return "Outono"
    elif (mes == 6 and dia >= 21) or (mes == 7) or (mes == 8) or (mes == 9 and dia < 23):
        return "Inverno"
    elif (mes == 9 and dia >= 23) or (mes == 10) or (mes == 11) or (mes == 12 and dia < 21):
        return "Primavera"

mes_numero = int(input("Digite o número do mês: "))
dia_numero = int(input("Digite o dia do mês: "))

for mes, numero in meses.items():
    if numero == mes_numero:
        estacao = get_estacao(numero, dia_numero)
        print(f"O mês é {mes} e a estação é {estacao}")